### DISCUSSION TITLE
DiscordSRV Alert not showing new town name?

### DISCUSSION THREAD

**Discussion by KarlosWGeek** [TownyAdvanced/Towny#7553] [Q&A] [answered]
I'm trying to create a DiscordSRV alert for creating a new town through Towny, but if I use the placeholder %townyadvanced_town% on the alert, it comes back empty! I'm thinking it's because the player that made the town wasn't part of a town right before making one, so it doesn't pull the town name because they didn't have one until right now.
Is there another placeholder I could use to get the name of the town that a player just created?

**Comment by LlmDl** [comment]
What event did you put in your alerts.yml.

Show me what you used to get an error.

**Reply by KarlosWGeek** [reply]
# Towny Events - New Town
  - Trigger: PreNewTownEvent
    Channel: towny-events
    Embed:
      Color: '#0062ff'
      Author:
        ImageUrl: "{embedavatarurl}"
        Name: "New Town Created"
      Description: "The town of **%townyadvanced_town%** has been created by **{username}**."
      Timestamp: true



The plan was to have an alert that goes like this (when a player creates a new town):
> The town of **NewTownName** has been created by **PlayerUsername**.

But the problem is that **%townyadvanced_town%** always comes out blank. My guess is that the player making the new town doesn't have a town at the moment the alert triggers, so it comes blank.

I also tried replacing **%townyadvanced_town%** with **${#event.getTown()}** and with **${#event.getTown().getFormattedName()}** but that also doesn't output anything.

**Comment by LlmDl** [comment, marked as answer]
Don't use PreNewTownEvent, use NewTownEvent. The Pre event is thrown before the town exists.

**Reply by LlmDl** [reply]
I also don't think you'll find that PAPI placeholders work in this case. Those use a Player to parse.
${#event.getTown()} is probably what you want to do.
